lib/render/SliceRenderer.cpp

```diff
@@ -0,0 +1,374 @@
+    double dy = (_cacheParams.boxMax[Y]-_cacheParams.boxMin[Y])/(1+sampleRate);
+    double dz = (_cacheParams.boxMax[Z]-_cacheParams.boxMin[Z])/(1+sampleRate);
+
+    if (_cacheParams.orientation == XY) {
+        coords[X] = _cacheParams.boxMin[X] + dx*i + dx/2.f;
+        coords[Y] = _cacheParams.boxMin[Y] + dy*j + dy/2.f;
+        coords[Z] = _cacheParams.boxMin[Z];
+    }
+    else if (_cacheParams.orientation == XZ) {
+        coords[X] = _cacheParams.boxMin[X] + dx*i + dx/2.f;
+        coords[Y] = _cacheParams.boxMin[Y];
+        coords[Z] = _cacheParams.boxMin[Z] + dz*j + dz/2.f;
+    }
+    else {  // Y corresponds to i, the faster axis; Z to j, the slower axis
+        coords[Z] = _cacheParams.boxMin[Z] + dz*j + dz/2.f;
+        coords[Y] = _cacheParams.boxMin[Y] + dy*i + dy/2.f;
+        coords[X] = _cacheParams.boxMin[X];
+    }
+}
+
+int SliceRenderer::_saveTextureData() {
+    Grid* grid = NULL;
+    int rc = DataMgrUtils::GetGrids(
+        _dataMgr, 
+        _cacheParams.ts,
+        _cacheParams.varName,
+        _cacheParams.boxMin,
+        _cacheParams.boxMax,
+        true,
+        &_cacheParams.refinementLevel,
+        &_cacheParams.compressionLevel,
+        &grid
+    );
+    
+    if (rc<0) {
+        return(-1);
+    }
+    assert(grid);
+
+    std::vector<double> textureMin, textureMax;
+    _getTextureCoordinates(textureMin, textureMax);
+
+    std::vector<double> cachedValuesForParams;
+
+    int missing=0;
+    float varValue, minValue, maxValue, missingValue; 
+    std::vector<double> coords(3, 0.0); 
+    for (int j=0; j<_textureHeight; j++) {
+        for (int i=0; i<_textureWidth; i++) {
+            _getSampleCoordinates(coords, i, j);
+
+            int index = (j*_textureWidth + i)*4;
+
+            varValue = grid->GetValue(coords);
+            missingValue = grid->GetMissingValue();
+            if (varValue == missingValue) {
+                missing++;
+                _textureData[index + 0] = 0.f;
+                _textureData[index + 1] = 0.f;
+                _textureData[index + 2] = 0.f;
+                _textureData[index + 3] = 0.f;
+                continue;
+            }
+
+            cachedValuesForParams.push_back(varValue);
+           
+            minValue = _cacheParams.tf_minMax[0];
+            maxValue = _cacheParams.tf_minMax[1];
+            int bin = 255 * (varValue-minValue) / (maxValue-minValue);
+            bin *= 4;
+            if (bin < 0)
+                bin = 0;
+            if (bin >= _cacheParams.tf_lut.size())
+                bin = _cacheParams.tf_lut.size()-4;
+            
+            unsigned char red, green, blue, alpha;
+            red   = _cacheParams.tf_lut[bin + 0]*255;
+            green = _cacheParams.tf_lut[bin + 1]*255;
+            blue  = _cacheParams.tf_lut[bin + 2]*255;
+            alpha = _cacheParams.tf_lut[bin + 3]*255;
+
+            _textureData[index + 0] = red;
```

Comment:
You should use an "indirect" texture for the slicer: store the data value in the texture and let the shader do a lookup into 
the color map. That way you don't have to rebuild the texture every time the color map changes. Also, the data can be stored directly as floats. See Sam's volume renderer for an example.

Alternatively, it might be acceptable to cache the sampled slice data and generate the texture from the cached data to avoid querying the grid each time the colormap changes. The Grid::GetValue method can be expensive.